packages/iris-grid/src/GotoRow.tsx

```diff
@@ -37,67 +45,58 @@ const GotoRow = ({
   onGotoRowNumberChanged,
   onClose,
 }: GotoRowProps): ReactElement => {
-  const [row, setRow] = useState('');
-  const [error, setError] = useState('');
+  const inputRef = useRef<HTMLInputElement>(null);
 
   const res = 'Row number';
 
   const { rowCount } = model;
 
+  useEffect(() => {
+    if (selectGotoRowInput) {
+      inputRef.current?.select();
+    }
+  }, [selectGotoRowInput]);
   return (
     <IrisGridBottomBar
       isShown={isShown}
       className={classNames('goto-row')}
       onEntering={onEntering}
-      onEntered={onEntered}
+      onEntered={() => {
+        onEntered();
+        inputRef.current?.focus();
+      }}
       onExiting={onExiting}
       onExited={onExited}
     >
       <>
-        <div className="goto-row-text">
-          <h6>Go to row</h6>
-        </div>
-        <div className="goto-row-input">
-          <input
-            type="number"
-            className={classNames('form-control', {
-              'is-invalid': error !== '',
-            })}
-            placeholder={res}
-            onChange={event => {
-              const rowNumber = event.target.value;
-              setRow(rowNumber);
-              if (rowNumber === '') {
-                setError('');
-                return;
-              }
-              const rowInt = parseInt(event.target.value, 10);
-              if (rowInt > rowCount || rowInt < -rowCount) {
-                setError('Invalid row index');
-              } else if (rowInt === 0) {
-                onGotoRowNumberChanged(1);
-                setError('');
-              } else if (rowInt < 0) {
-                onGotoRowNumberChanged(rowInt + rowCount + 1);
-                setError('');
-              } else {
-                onGotoRowNumberChanged(parseInt(event.target.value, 10));
-                setError('');
-              }
-            }}
-            value={row}
-          />
-        </div>
-        <div className="goto-row-text">
-          <h6>
+        <div className="goto-row-wrapper">
+          <div className="goto-row-text">Go to row</div>
+          <div className="goto-row-input">
+            <input
+              ref={inputRef}
+              type="number"
+              onKeyDown={e => {
+                if (e.key === 'Enter') {
+                  onSubmit();
```

Comment:
You should consume this event
```suggestion
                  e.stopPropagation();
                  e.preventDefault();
                  onSubmit();
```